src/pubd/repository.rs

```diff
@@ -179,9 +176,14 @@ impl RepositoryContentProxy {
         self.read(|content| content.write_repository(config))
     }
 
-    /// Reset the RRDP session
+    /// Reset the RRDP session if is initialized. Otherwise do nothing.
     pub fn session_reset(&self, config: &RepositoryRetentionConfig) -> KrillResult<()> {
-        self.write(|content| content.session_reset(config))
+        if self.cache.read().unwrap().is_some() {
```

Comment:
Is `unwrap()` correct here or should that be the `?` operator?

Reply:
This can only fail because of poison errors. And we can only have those in case there was a panic while a lock was hold. Krill is set up to exit on panic. So, this is safe and commonly done. Furthermore.. even if Krill did not crash earlier it should crash now if a poison error would occur because we cannot recover from it.